src/cacheRefresher.js

```diff
@@ -0,0 +1,185 @@
+                    lastModified: this._lastTime,
+                    [this._cacheSchema]: {}
+                }
+            }
+
+            var xmlDoc = DomUtil.fromJSON("cache", jsonCache, 'SimpleJson');
+            soapCall.writeDocument("script", xmlDoc);
+
+            // Do a soap call GetModifiedEntities instead of xtksession.GetModifiedEnties because we don't want to go through methodCache 
+            // which can be wrong just after a build updgarde from a old version of acc that has not the method GetModifiedEntities and 
+            // a new version of acc that has the method GetModifiedEntities
+            return this._client._makeSoapCall(soapCall)
+                .then(() => {
+                    var doc = soapCall.getNextDocument();
+                    soapCall.checkNoMoreArgs();
+                    doc = that._client._toRepresentation(doc, 'xml');
+                    that._lastTime = DomUtil.getAttributeAsString(doc, "time"); // save time to be able to send it as an attribute in the next soap call
+                    that._buildNumber = DomUtil.getAttributeAsString(doc, "buildNumber");
+                    that.refresh(doc, that._cacheSchema);
+                    that._metadataCache.put("time", that._lastTime);
+                    that._metadataCache.put("buildNumber", that._buildNumber);
+                    Promise.resolve();
+                })
+                .catch((ex) => {
+                    if (soapCall.methodName == "GetModifiedEntities" && ex.errorCode == "SOP-330006") {
+                        clearInterval(this._intervalId);
+                        this._intervalId = null;
+                    }
+                });
+        }
+
+        // Refresh Cache : remove entities modified recently listed in xmlDoc
+        refresh(xmlDoc, cacheSchema) {
+            console.log("cache refresh " + cacheSchema);
+            const bClearCache = XtkCaster.asBoolean(DomUtil.getAttributeAsString(xmlDoc, "emptyCache"));
+            if (bClearCache == true) {
+                console.log("Clear cache");
+                this._cache.clear();
+            } else {
+                var child = DomUtil.getFirstChildElement(xmlDoc, "entityCache");
+                while (child) {
+                    let schemaId = DomUtil.getAttributeAsString(child, "pk");
+                    let schemaType = DomUtil.getAttributeAsString(child, "schema");
+                    if (schemaType == cacheSchema) {
+                        console.log("remove " + schemaId); // TODO: delete log
+                        this._cache.remove(schemaId);
+                    }
+                    child = DomUtil.getNextSiblingElement(child);
+                }
+            }
+        }
+
+        stopAutoRefresh() {
+            clearInterval(this._intervalId);
+            this._intervalId = null;
+        }
+    }
+
+    // Public exports
+    exports.CacheRefresher = CacheRefresher;
+    exports.MetadataCache = MetadataCache;
```

Comment:
MetadataCache should be private to the cacheRefresher file, and hence not exported